backend/src/controllers/group/get-all-groups.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response } from "express";
+import { Like } from "typeorm";
+
+import dataSource from "../../data-source";
+import { Group } from "../../entity/group.entity";
+
+const getAllGroups = async (req: Request, res: Response) => {
+  const groupRepository = dataSource.getRepository(Group);
+
+  try {
+    const { filterByTitle, filterByContent } = req.query;
+
+    if (filterByTitle != null || filterByContent != null) {
+      const groups = await groupRepository.findBy({
+        title: Like(`%${filterByTitle ?? ""}%`),
+        content: Like(`%${filterByContent ?? ""}%`),
+      });
+      res.status(200).json({
+        status: "success",
+        data: {
+          groups,
+        },
+      });
+      return;
```

Comment:
You can do the return right away, like so:
```suggestion
      return res.status(200).json({
        status: "success",
        data: {
          groups,
        },
      });
```